File: code/transform/features.py
class Features():
    def __init__(self, frames):
        self.leftarm = Ranges()
        self.rightarm = Ranges()
        for frame in frames:
            self.leftarm.update(frame.leftarm)
            self.rightarm.update(frame.rightarm)

    def dump(self):
        print 'Left Arm:'
        self.leftarm.dump()
        print 'Right Arm:'
        self.rightarm.dump()

class Ranges():
    def __init__(self):
        self.minx = self.miny = self.minz = 9999.0
        self.maxx = self.maxy = self.maxz = -9999.0

    def update(self, bodypart):
        self.minx = min(self.minx, bodypart.tx)
        self.maxx = max(self.maxx, bodypart.tx)
        self.miny = min(self.miny, bodypart.ty)
        self.maxy = max(self.maxy, bodypart.ty)
        self.minz = min(self.minz, bodypart.tz)
        self.maxz = max(self.maxz, bodypart.tz)

    def dump(self):
        dx = self.maxx - self.minx
        dy = self.maxy - self.miny
        dz = self.maxz - self.minz
        print 'dx = %f, dy = %f, dz = %f' % (dx,dy,dz)

File: code/transform/Features.java
import java.util.*;

class Features
{
	Ranges leftarm, rightarm;
	
	Features(ArrayList<Frame> data)
	{
		leftarm = new Ranges();
		rightarm = new Ranges();
		for(Frame f: data)
		{
			leftarm.update(f.left);
			rightarm.update(f.right);
		}
	}
	
	void dump()
	{
		System.out.println("Left Arm:");
		leftarm.dump();
		System.out.println("Right Arm:");
		rightarm.dump();
	}
};

class Ranges
{
	double minx, maxx, miny, maxy, minz, maxz;
	
	Ranges()
	{
		minx = miny = minz = 9999.0;
		maxx = maxy = maxz = -9999.0;
	}

	void update(SixDOF bodypart)
	{
		minx = Math.min(minx, bodypart.tx);
		maxx = Math.max(maxx, bodypart.tx);
		miny = Math.min(miny, bodypart.ty);
		maxy = Math.max(maxy, bodypart.ty);
		minz = Math.min(minz, bodypart.tz);
		maxz = Math.max(maxz, bodypart.tz);
	}

	void dump()
	{
		double dx = maxx - minx;
		double dy = maxy - miny;
		double dz = maxz - minz;
		System.out.printf("dx = %f, dy = %f, dz = %f\n", dx, dy, dz);
	}
};
